src/Workspaces/Core/Portable/Classification/ClassifierHelper.cs

```diff
@@ -198,10 +198,18 @@ private static void AdjustSpans(ArrayBuilder<ClassifiedSpan> spans, TextSpan wid
                 // Take the semantic part if it's just 'text'.  We want to keep it if
                 // the semantic classifier actually produced an interesting result 
                 // (as opposed to it just being a 'gap' classification).
-                var part = replacementIndex >= 0 && !IsClassifiedAsText(semanticParts[replacementIndex])
-                    ? semanticParts[replacementIndex]
-                    : syntaxPartAndSpan;
-                finalParts.Add(part);
+                if (replacementIndex >= 0 && !IsClassifiedAsText(semanticParts[replacementIndex]))
+                {
+                    finalParts.Add(semanticParts[replacementIndex]);
+                }
+                // We might already have a semantic part for the given TextSpan, in
+                // which case we don't want to add the syntactic part unless it's a
+                // modifier.
```

Comment:
nit: this says modifier, but the lookup is on additive type names. Are there additive type names that aren't modifiers? Can we use the same terminology here to be consistent?

Reply:
Currently there's only one additive type name (static), but I'll change the comment to use consistent terminology. Thanks for the feedback!